src/Factory.php

```diff
@@ -110,16 +111,14 @@ private function parseUrl($target)
             $parts['host'] = '127.0.0.1';
         }
 
-        $auth = null;
-        if (isset($parts['user'])) {
-            $auth = $parts['user'];
-        }
+        // username:password@ (Redis doesn't support usernames)
         if (isset($parts['pass'])) {
-            $auth .= ':' . $parts['pass'];
-        }
-        if ($auth !== null) {
-            $parts['auth'] = $auth;
-        }
+			$parts['auth'] = $parts['pass'];
+		}
+		// password@
+		else if (isset($parts['user'])) {
+			$parts['auth'] = $parts['user'];
+		}
```

Comment:
Can you fix the coding style here? E.g. `} else if (...) {` and fix the indentation.